charts/consul/templates/connect-inject-deployment.yaml

```diff
@@ -135,6 +135,11 @@ spec:
                 {{- else }}
                 -default-enable-transparent-proxy=false \
                 {{- end }}
+                {{- if .Values.connectInject.cni.enabled }}
+                -enable-cni=true \
+                {{- else }}
+                --enable-cni=false \
+                {{- end }}
```

Comment:
If you default the flag to `false` in `command.go` you could get away with just passing `-enable-cni=true` when it is enabled.
In either case you've got a typo on the `-enable-cni=false` line (2 hyphens)

Reply:
I do have the command set to false by default but I was following the `if .Values.connectInject.transparentProxy.defaultEnabled ` pattern above (and just copying it).